src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,82 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = new Node<>(key, value, null);
+                size++;
+                return;
+            }
+        }
+        table[index] = new Node<>(key, value, null);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+        Node<K, V>[] oldTable = table;
+        table = new Node[table.length * 2];
```

Comment:
ты сначала пересчитываешь threshold, а потом меняешь table.length, так быть не должно, пример:
 - изначально threshold == 12
 - первый раз попадаем в resize, пересчитываем threshold, опять 12 
 - меняем размер массива
 - следующий раз, когда мы вызываем put, мы опять попадаем в resize, и только на воторой раз пересчитываем
```suggestion
        Node<K, V>[] oldTable = table;
        table = new Node[table.length * 2];
        threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
```